lib/libvarnish/vtim.c

```diff
@@ -179,6 +181,70 @@ VTIM_format(vtim_real t, char *p)
 	    tm.tm_hour, tm.tm_min, tm.tm_sec));
 }
 
+void
+VTIM_format_web(vtim_real t, char p[VTIM_FORMAT_WEB_SIZE])
+{
+	char buf[sizeof "0.123456"], *z;
+	struct vsb vsb[1];
+	struct tm tm;
+	time_t tt;
+	double frac, i;
+
+	AN(p);
+	*p = '\0';
+
+	if (t < (vtim_real)INTMAX_MIN || t > (vtim_real)INTMAX_MAX)
+		return;
+
+	tt = (time_t)(intmax_t)t;
+	if (gmtime_r(&tt, &tm) == NULL)
+		return;
+
+	AN(VSB_init(vsb, p, VTIM_FORMAT_WEB_SIZE));
+	VSB_printf(vsb, "%04d-%02d-%02dT%02d:%02d:%02d",
+	    tm.tm_year + 1900, tm.tm_mon, tm.tm_mday,
+	    tm.tm_hour, tm.tm_min, tm.tm_sec);
+
```

Comment:
Why even bother with the VSB ?

You are printing a fixed-length format, use strftime() directly ?

Reply:
Because what follows is of variable length.